util/src/main/java/org/datacommons/util/SummaryReportGenerator.java

```diff
@@ -0,0 +1,99 @@
+package org.datacommons.util;
+
+import freemarker.template.Configuration;
+import freemarker.template.Template;
+import freemarker.template.TemplateException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+import org.datacommons.proto.Debug;
+
+public class SummaryReportGenerator {
+
+  // An object to save the information about a stat var. This contains all the necessary getters to
+  // access the information in this object from SummaryReport.ftl
+  public static final class StatVarSummary {
```

Comment:
It is somewhat strange that parts of this is updated in the fast-path and parts of it used just for rendering. I wonder if its worth separating them... maybe a TODO for this PR?

Reply:
The thing is that the parts that are updated in the fast-path are saved to be read later for rendering .. but I can still add a TODO to look into this class to figure out a better way to handle everything